becquerel/core/spectrum.py

```diff
@@ -1070,16 +1070,13 @@ def get_bin_properties(self, use_kev=None):
             return self.bin_edges_raw, self.bin_widths_raw, self.bin_centers_raw
 
     def apply_calibration(self, cal):
-        """Use an EnergyCal to generate bin edge energies for this spectrum.
+        """Use a Calibration to generate bin edge energies for this spectrum.
 
         Args:
-          cal: an object derived from EnergyCalBase
+          cal: a Calibration object
         """
 
-        try:
-            self.bin_edges_kev = cal.ch2kev(self.bin_edges_raw)
-        except AttributeError:
-            self.bin_edges_kev = cal(self.bin_edges_raw)
```

Comment:
Should we maintain backwards-compatibility for a while?

Reply:
Yeah, probably, I was too zealous. I'll restore the old ways and throw a `DeprecationWarning` when they are used.